### ISSUE TITLE
Feature Request: Allow for smaller window.

### ISSUE THREAD

**Issue by M0JXD** [Stazed/non-mixer-xt#47]
I've been using Non a lot more recently and I've found having the windows for the timeline and mixer next to each other is quite a nice way of working:

![smaller_window](https://github.com/user-attachments/assets/4f867bfd-f7d9-4277-b266-fbec4d255068)

Sometimes I only do stuff with a couple of tracks (backing track and a guitar sort of thing) so being able to make the mixer smaller would be useful for me :)

Thanks as always!

**Comment by M0JXD** [comment]
I've just noticed the title is missing for Timeline in that picture, I've just opened my session again and it's there. Maybe a bug although I think I'd just renamed the session so perhaps not.

**Comment by Stazed** [comment]
The missing project name is an NTK bug. Seems to happen often when the transport is rolling and auto scroll is going. Does not happen with FLTK. Wouldn't know where to start in looking for this... You can get it back by switching to another workspace and back also. wip branch as change for smaller window.

**Comment by M0JXD** [comment]
Thanks - although CMake can't find NTK anymore for me:

![image](https://github.com/user-attachments/assets/5baacb75-d3fa-49ed-b4b1-363bd2d237c9)

It's definitely under usr/local/lib
I see you changed the CMakelist just a few days ago for the wip branch. What might be missing is NTK headers under usr/local/include/ntk

I had no end of trouble getting xmake to register NTK (and sigc++) in my sequencer port and it was because I needed to force the build system to find the .a/.so objects and the headers.

**Comment by M0JXD** [comment]
I've built for FLTK and looking good! Thanks :)

![image](https://github.com/user-attachments/assets/45905ea5-d35d-445f-9b5a-cbddc7620551)

**Comment by Stazed** [comment]
Oh, forgot to mention, moved NTK to a static build like FLTK as it is really necessary to avoid conflicts with certain plugins.
Check out the README of my "unofficial" NTK repository for the various incantations to get it to find and link, etc. Kind of a bitch...

**Comment by M0JXD** [comment]
Thanks, setting the env variables has allowed it to build but CMake still doesn't detect NTK's existence no matter how I set PKG_CONFIG_PATH and LD_LIBRARY_PATH. I'm still using old NTK without your latest changes.

**Comment by Stazed** [comment]
>...setting the env variables has allowed it to build but CMake still doesn't detect NTK's existence...

You say it builds, but doesn't detect NTK's existance?? Not sure what you mean. 

Assuming install is to /usr/local/lib
Here are the steps that have worked for me in terminal:

1)  export PKG_CONFIG_PATH=/usr/local/lib/pkgconfig:$PKG_CONFIG_PATH
2)  export LD_LIBRARY_PATH=/usr/local/lib (Only necessary for runtime error)
3)  export CPLUS_INCLUDE_PATH="/usr/local/include/ntk/:$CPLUS_INCLUDE_PATH"
4)  export C_INCLUDE_PATH="/usr/local/include/ntk/:$C_INCLUDE_PATH"

Then in a new build directory from the same terminal:
cmake -DEnableNTK=ON ..
cmake -j

The key here may be that if the build previously failed, you need to delete everything in the build folder to start fresh or the cmake cache will have old information.

**Comment by M0JXD** [comment]
Yep it builds, but CMake still lists the library as not found.

Without exports we get the expected, it tries to link to FLTK and throws an error when it hits the transparency difference in NTK:
![mix1](https://github.com/user-attachments/assets/4864ff07-f07c-47db-a689-58df1ef1f010)
![mix2](https://github.com/user-attachments/assets/469ed5b9-0a0f-49aa-a688-4d7195d3ff98)

If I add the exports as mentioned, CMake still says it can't be found, but make runs fine and links to NTK:

![mix3](https://github.com/user-attachments/assets/7b875a7c-6670-410f-be2f-2c4766a80afe)
![mix4](https://github.com/user-attachments/assets/142a2455-bccb-4b47-841e-8724fa72ebf4)
![mix5](https://github.com/user-attachments/assets/ae4b0cb4-5c11-4b47-a34a-1240ad31806b)

**Comment by Stazed** [comment]
O.K. was a display issue. Is fixed now in wip.

**Comment by M0JXD** [comment]
Thank you, I haven't a clue regarding CMake 😅

**Comment by M0JXD** [comment]
As I've been using this I don't think it's restoring the window sizes correctly when the window is smaller than what was previously allowed.

**Comment by Stazed** [comment]
Yep, I broke NSM window saving in some cases when I re-factored things a little for this feature. Should be fixed now.
Thanks for reporting!

**Comment by M0JXD** [comment]
Super, thank you :)
I can't reiterate enough how much I appreciate all you're doing on this!

**Comment by M0JXD** [comment]
I just built your WIP branch (need to set up a cron job) and the env variables caught me out again. Is there any way this can be set automatically by CMake?

Also, great work on NTK-Unofficial :)

**Comment by Stazed** [comment]
Cleaned up the static NTK build so that CPLUS_INCLUDE_PATH, C_INCLUDE_PATH are not needed.
The damn manjaro distro NTK install is missing a file needed for static linking which was 
confusing me, grrrr!!!

Digging into this library stuff, it seems that installing to /usr/local/ is not generally supported
by default with most distros. Doing so will usually require setting PKG_CONFIG_PATH and LD_LIBRARY_PATH
and I have found can cause some runtime problems. Because of this I do not want to support any /usr/local/ install 
of NTK.

I have updated the installation instructions in the ntk-unofficial README for installing to /usr/.

Installing NTK to /usr/ means all that is needed for NTK build is:

mkdir build
cd build
cmake -DEnableNTK=ON ..
make
sudo make install

**Comment by M0JXD** [comment]
The benefit of usr/local is that it's generally untouched by the package manager but yes I can see why it's a problem (why am I not surprised Manjaro is a problematic one? 😅). I pretty much exclusively use Debian based distros which have never packaged NTK so it's not a problem installing to usr for me.

I think provided it's mentioned in the README it installs to usr by default due to these issues (as this is atypical of source installations) and a mention of the possibly required environment variable exports when using "-DCMAKE_INSTALL_PREFIX=/usr/local" should suffice.

**Comment by M0JXD** [comment]
Edit: Sorry for my stupidity, NTK-Unofficial still uses waf (unless you intend to change that?) Same applies though and I see it's now mentioned :)
